### DISCUSSION TITLE
Questions: In which cases will "goodVendorConfiguration" need to change, and what will happen till it changes?

### DISCUSSION THREAD

**Discussion by AndroidDeveloperLB** [bocops/UMP-workarounds#12] [Q&A]
So suppose I have Admob (or a different ad provider/mediation), together with some ad providers as part of mediation (or none).

It says I need to fill the "goodVendorConfiguration":

https://github.com/bocops/UMP-workarounds/blob/main/detect_insufficient_consent/android/kotlin/detect_insufficient_consent.kt#L72

```
    // TODO vendor configuration is variable, so needs to be defined by the individual developer
    //   - run app and make sure that a valid configuration is stored
    //   - have the app log the value of [tcVendorString], then copy that value to the following line
    //   - repeat if ad configuration changes, perhaps make this value available via remote configuration instead
    val goodVendorConfiguration = "TODO"
```

Suppose I do fill it and right now it works fine, and can detect "All" vs other cases, meaning I can detect that the user has accepted to all ad-consent options.
I know it says we should have remote-config for this, but there are other issues as to how we even detect we need to change it, and what happens till we do notice it.

My questions:

1. what exactly will make it need a change? Will adding a new ad-source for the mediation mean it will need to change? Or maybe need to change something in the list of vendors? Or maybe even Google could change it without me noticing or doing anything to trigger it?
I ask to know how reliable is this method.

2. What will happen when the goodVendorConfiguration doesn't match perfectly ? Is it possible to have "Maybe consent-all" ? 

3. On StackOverflow, there is a similar detection method but it talks only about Admob:
https://stackoverflow.com/a/68310602/878126
What happens there? Will it stay "ok" as long as we use Admob (as mediation and/or ad-source) ? Meaning that we know that at least for Admob, it should be fine to get ads? 
I ask this because I know that limited ads are problematic for Admob?

**Comment by bocops** [comment]
I understand that this is important to you - but you will have to understand that I can't keep explaining the same things to you forever. We've been through most of this already, and you should be able to figure out the rest from the code here and other resources that you already have access to. If I haven't managed to make myself clear until now, I'm sure this won't change with yet another reply with the same content.

For what it is worth, there's a new Discord server making the rounds. Maybe the people over there can explain things better than I did. Maybe they will just tell you the same things. In any case, here's the link: https://discord.com/invite/5UnXR4fa

**Comment by AndroidDeveloperLB** [comment]
We haven't talked about these exact questions.
And the repository wasn't updated to explain those either, including in the comment for the "TODO".
This information should be publicly available in order to use this function well...
I need to work on this and there are only 2 places I've found that handle it in any way. 
The documentation of Google isn't clear about this and takes a lot of pages too, and it's quite hard to test this subject to learn about it.
Am I banned ? If so, please re-consider from banning me... I didn't mean to upset anyone.
I tried to use the repository and also reported about bugs, including a crash...
I don't use Discord. Why does it matter where the talk will be?
